pydantic_factories/factory.py

```diff
@@ -437,6 +443,10 @@ def build(cls, **kwargs) -> T:
                     kwargs[field_name] = cls.handle_factory_field(field_name=field_name)
                 else:
                     kwargs[field_name] = cls.get_field_value(model_field=model_field)
+        if factory_use_construct:
+            if is_pydantic_model(cls.__model__):
+                return cls.__model__.construct(**kwargs)  # type: ignore
```

Comment:
rather than using a type: ignore comment, please use `cast`:

```suggestion
                return cast(BaseModel, cls.__model__).construct(**kwargs)
```

Reply:
actually, this still fails mypy
 error: Incompatible return value type (got "BaseModel", expected "DataclassProtocol")
 should I add back the type:  ignore?